langchain/utils.py

```diff
@@ -19,3 +24,46 @@ def get_from_dict_or_env(
             f" `{env_key}` which contains it, or pass"
             f"  `{key}` as a named parameter."
         )
+
+
+
+def retry(ExceptionToCheck: Type[Exception] = Exception, tries: int = 5, delay: float = 0.5, backoff: int = 2, ExceptionToRaise: Type[Exception] =AssertionError) -> Callable:
```

Comment:
for variable names, can we do `exception_to_check`, etc?

also, for this specific usage can we pass in the openai rate limit error class

Reply:
Sure happy to rework it to match style. 

For the default openapi exceptions to catch, I regularly see RateLimitError and ServiceUnavailableError. This catches those by default along with other random network issues.   Do you want to limit the retry to a certain class of errors, or exclude certain errors?  

Also today I started running into weird RateLimitErrors that required more aggressive backoffs.  (I think openai token rate limiting logic is currently broken.) We may need to increase the default number of tries beyond  5.